llama_index/llm_predictor/base.py

```diff
@@ -169,9 +169,13 @@ def __init__(
         retry_on_throttling: bool = True,
         cache: Optional[BaseCache] = None,
         callback_manager: Optional[CallbackManager] = None,
+        ensure_llm: bool = True,
     ) -> None:
         """Initialize params."""
-        self._llm = llm or OpenAI(temperature=0, model_name="text-davinci-003")
+        self._llm = llm
+        if not self._llm and ensure_llm:
+            self._llm = OpenAI(temperature=0, model_name="text-davinci-003")
```

Comment:
It becomes confusing if Vellum acts as an LLM, but then there's a non-null value for `self._llm` which is an instance of `BaseLanguageModel`. I looked into trying to subclass `BaseLanguageModel` and make a `VellumLanguageModel` to act as a dummy/proxy, but this didn't seem tenable.

Reply:
I wonder if a better approach here is to create a `BasePredictor` class, which both `LLMPredictor` and `VellumPredictor` both sublasses.

This way, logic specific to LangChain integration could live in `LLMPredictor`.

Reply:
If you think this is a reasonable idea (given how Vellum differs in interface), I can assist in making this refactor.

Reply:
I think this is a great idea and was my preferred approach, but was mindful of introducing too many invasive changes in this first POC.